test/e2e/framework/pod/wait.go

```diff
@@ -830,18 +830,3 @@ func handleWaitingAPIError(err error, retryNotFound bool, taskFormat string, tas
 	framework.Logf("Encountered non-retryable error while %s: %v", taskDescription, err)
 	return false, err
 }
-
-// Decide whether to retry an API request. Optionally include a delay to retry after.
-func shouldRetry(err error) (retry bool, retryAfter time.Duration) {
```

Comment:
I´ve already mentioned my concern when we added this , since client-go already implements this logic internally, adding 10 retries by default

Reply:
So you want to reopen the discussion around this?

I thought it had been settled when merging https://github.com/kubernetes/kubernetes/pull/109704 and https://github.com/kubernetes/kubernetes/commit/55ec0f2d9dd7a86dc1a487b035e725fd9d12721e.

We would have to pull in Tim again because I don't have a strong opinion on this and this particular logic is what he wanted.

Reply:
> we have a 2 loops now that retry x*10 on the same error

Just because we have two nested loops does not necessarily mean that there are more API requests. That depends on the delays that each loop is using. We don't control how quickly client-go retries.

What we decide here is whether we want to continue retrying for the entire duration of the overall timeout.

Reply:
https://github.com/kubernetes/kubernetes/pull/109816#issuecomment-1120892122

Reply:
:/ https://github.com/kubernetes/kubernetes/pull/111555

Reply:
> We don't control how quickly client-go retries.

every second https://github.com/kubernetes/kubernetes/pull/109816#issuecomment-1119332876

my point is that I don't think that a test that has to retry for 1 minute should pass

Reply:
but I'm not super strong just wanted to try another push XD